src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -1195,125 +1194,115 @@ namespace SurfaceGeometry {
         // Debug    write(outputfiledebug,*) ' need to add ',NeedtoAddSurfaces+NeedToAddSubSurfaces
         for (int SurfNum = 1; SurfNum <= FirstTotalSurfaces; ++SurfNum) {
             auto &surfTemp = state.dataSurfaceGeometry->SurfaceTmp(SurfNum);
-            if (surfTemp.ExtBoundCond != UnenteredAdjacentZoneSurface) continue;
+            if ((surfTemp.ExtBoundCond != unenteredAdjacentZoneSurface) && (surfTemp.ExtBoundCond != unenteredAdjacentSpaceSurface)) continue;
             // Need to add surface
             ++CurNewSurf;
             // Debug    write(outputfiledebug,*) ' adding surface=',curnewsurf
-            state.dataSurfaceGeometry->SurfaceTmp(CurNewSurf) = state.dataSurfaceGeometry->SurfaceTmp(SurfNum);
+            auto &newSurf = state.dataSurfaceGeometry->SurfaceTmp(CurNewSurf);
+            newSurf = surfTemp;
             //  Basic parameters are the same for both surfaces.
-            Found = Util::FindItemInList(surfTemp.ExtBoundCondName, state.dataHeatBal->Zone, state.dataGlobal->NumOfZones);
-            if (Found == 0) continue;
-            state.dataSurfaceGeometry->SurfaceTmp(CurNewSurf).Zone = Found;
-            state.dataSurfaceGeometry->SurfaceTmp(CurNewSurf).ZoneName = state.dataHeatBal->Zone(Found).Name;
+            if (surfTemp.ExtBoundCond == unenteredAdjacentZoneSurface) {
+                int Found = Util::FindItemInList(surfTemp.ExtBoundCondName, state.dataHeatBal->Zone, state.dataGlobal->NumOfZones);
+                if (Found == 0) continue;
+                newSurf.Zone = Found;
+                auto &newZone = state.dataHeatBal->Zone(Found);
+                newSurf.ZoneName = newZone.Name;
+                assert(newZone.spaceIndexes.size() >= 1);
+                newSurf.spaceNum = 0; // clear this here and set later
+            } else if (surfTemp.ExtBoundCond == unenteredAdjacentSpaceSurface) {
+                int Found = Util::FindItemInList(surfTemp.ExtBoundCondName, state.dataHeatBal->space, state.dataGlobal->numSpaces);
+                if (Found == 0) continue;
+                newSurf.spaceNum = Found;
+                int zoneNum = state.dataHeatBal->space(Found).zoneNum;
+                newSurf.Zone = zoneNum;
+                newSurf.ZoneName = state.dataHeatBal->Zone(zoneNum).Name;
```

Comment:
For space BC, set `spaceNum` here and locate the zone for that space.
The remaining changes in this function are just from the `newSurf` shortcut.

Reply:
And it would actually be nice if you could add a pattern match on GitHub's UI to ignore some changes.  You can already ignore whitespace-only changes, so the capability is somewhat in place.  If you could just say "ignore any changes that are purely from "state.dataSurfaceGeometry->SurfaceTmp(CurNewSurf)" to "newSurf", that would be incredible.

FWIW I think they are using the --ignore-all-space or whatever argument with Git Diff.  There are other diff arguments that could be leveraged (-G along with --pickaxe-regex).  OK I'm done I'm done.